src/modules/Chat.ts

```diff
@@ -0,0 +1,44 @@
+import { User } from "src/modules/User";
+import { Socket } from "socket.io-client";
+
+interface messageData {
+  message: string;
+  user: User;
+}
+export class Chat {
+  private user: User;
+  private roomId: string;
+  private channel: Socket;
+
+  constructor(user: User, roomId: string, channel: Socket) {
+    this.user = user;
+    this.roomId = roomId;
+    this.channel = channel;
+  }
+
+  sendMessage(message: string) {
+    this.channel.emit("sendMessage", {
+      message,
+      roomId: this.roomId,
+      user: this.user,
+    });
+  }
+
+  async fetchMessages(): Promise<messageData> {
+    this.channel.emit("requestAllMessages", {
+      roomId: this.roomId,
+      user: this.user,
```

Comment:
I don't think we need to send back the user object for this the roomId should be enough.